src/components/CloudinaryInput.tsx

```diff
@@ -17,24 +17,64 @@ const CloudinaryInput = (props: ObjectInputProps) => {
   const handleSelect = useCallback(
     (payload: InsertHandlerParams) => {
       const [asset] = payload.assets
+
       if (!asset) {
         return
       }
 
-      onChange(
-        PatchEvent.from([
-          set(
-            Object.assign(
-              {
-                _type: type.name,
-                _version: 1,
-                ...(value?._key ? {_key: value._key} : {_key: nanoid()}),
-              },
-              asset
-            )
-          ),
-        ])
-      )
+      //The cloudinary metadata (context) needs to be transformed to a valid object key
+      //because the key can't contain special characters in the Studio
+      if (asset.context) {
+        const customKey = Object.keys(asset.context.custom)[0]
```

Comment:
Do we know there can ever be only one key in this object?

Reply:
Was thinking the same thing.

Could we consider writing this as some version of 


```ts
const remappedEntries = Object.entries(asset.context.custom)
.map((key, value) => /* below code, probably extracted to a function for clarity */)

const updatedCustom = Object.fromEntries(remappedEntries)


const updatedAsset = {
          ...asset,
          context: {
            ...asset.context,
            custom: updatedCustom,
          },
        }
```

That should handle 0-n custom entries

Reply:
Ahh it sure can have several keys! Made a new commit and tested in the studio with several keys